tests/src/test/scala/common/BaseWsk.scala

```diff
@@ -0,0 +1,406 @@
+              since: Option[Instant],
+              retries: Int,
+              pollPeriod: Duration = 1.second)(implicit wp: WskProps): Seq[String]
+
+  def waitForActivation(activationId: String, initialWait: Duration, pollPeriod: Duration, totalWait: Duration)(
+    implicit wp: WskProps): Either[String, JsObject]
+
+  def get(activationId: Option[String] = None,
+          expectedExitCode: Int = SUCCESS_EXIT,
+          fieldFilter: Option[String] = None,
+          last: Option[Boolean] = None)(implicit wp: WskProps): RunResult
+
+  def console(duration: Duration, since: Option[Duration] = None, expectedExitCode: Int = SUCCESS_EXIT)(
+    implicit wp: WskProps): RunResult
+
+  def logs(activationId: Option[String] = None, expectedExitCode: Int = SUCCESS_EXIT, last: Option[Boolean] = None)(
+    implicit wp: WskProps): RunResult
+
+  def result(activationId: Option[String] = None, expectedExitCode: Int = SUCCESS_EXIT, last: Option[Boolean] = None)(
+    implicit wp: WskProps): RunResult
+
+}
+
+trait BaseNamespace extends RunWskCmd {
+
+  def list(expectedExitCode: Int = SUCCESS_EXIT, nameSort: Option[Boolean] = None)(implicit wp: WskProps): RunResult
+
+  def whois()(implicit wskprops: WskProps): String
+
+  def get(namespace: Option[String] = None, expectedExitCode: Int, nameSort: Option[Boolean] = None)(
+    implicit wp: WskProps): RunResult
+}
+
+trait BaseApi extends RunWskCmd {
+
+  def create(basepath: Option[String] = None,
+             relpath: Option[String] = None,
+             operation: Option[String] = None,
+             action: Option[String] = None,
+             apiname: Option[String] = None,
+             swagger: Option[String] = None,
+             responsetype: Option[String] = None,
+             expectedExitCode: Int = SUCCESS_EXIT,
+             cliCfgFile: Option[String] = None)(implicit wp: WskProps): RunResult
+
+  def list(basepathOrApiName: Option[String] = None,
+           relpath: Option[String] = None,
+           operation: Option[String] = None,
+           limit: Option[Int] = None,
+           since: Option[Instant] = None,
+           full: Option[Boolean] = None,
+           nameSort: Option[Boolean] = None,
+           expectedExitCode: Int = SUCCESS_EXIT,
+           cliCfgFile: Option[String] = None)(implicit wp: WskProps): RunResult
+
+  def get(basepathOrApiName: Option[String] = None,
+          full: Option[Boolean] = None,
+          expectedExitCode: Int = SUCCESS_EXIT,
+          cliCfgFile: Option[String] = None,
+          format: Option[String] = None)(implicit wp: WskProps): RunResult
+
+  def delete(basepathOrApiName: String,
+             relpath: Option[String] = None,
+             operation: Option[String] = None,
+             expectedExitCode: Int = SUCCESS_EXIT,
+             cliCfgFile: Option[String] = None)(implicit wp: WskProps): RunResult
+}
+
+trait RunWskCmd extends Matchers {
+
+  /**
+   * The base command to run.
+   */
+  def baseCommand = Wsk.baseCommand
```

Comment:
Here you make a call out to the Wsk singleton object that is defined in Wsk.scala. Since this file is the base for Wsk to inherit from, I don't think it should reference Wsk.scala at all.